src/message/messagesActions.js

```diff
@@ -21,24 +18,8 @@ export const doNarrow = (narrow: Narrow, anchor: number = FIRST_UNREAD_ANCHOR) =
     return;
   }
 
-  dispatch({ type: FETCH_STATE_RESET });
-
-  const allNarrows = getAllNarrows(state);
-  const messages = getMessages(state);
-  const messagesForNarrow = (allNarrows[JSON.stringify(narrow)] || NULL_ARRAY).map(
-    id => messages[id],
-  );
-  const tooFewMessages = messagesForNarrow.length < config.messagesPerRequest / 2;
-
-  const caughtUp = state.caughtUp[JSON.stringify(narrow)] || NULL_CAUGHTUP;
-  const isCaughtUp = caughtUp.newer && caughtUp.older;
-
-  if (anchor === FIRST_UNREAD_ANCHOR && tooFewMessages && !isCaughtUp) {
-    dispatch(fetchMessagesAtFirstUnread(narrow));
-  } else if (anchor !== FIRST_UNREAD_ANCHOR) {
-    dispatch(fetchMessagesAroundAnchor(narrow, anchor));
-  }
-
+  dispatch({ type: DO_NARROW, narrow });
+  dispatch(fetchMessagesInNarrow(narrow));
```

Comment:
This needs to pass `anchor`; otherwise the `anchor` passed to `doNarrow` gets dropped, and this fetch always uses the default.